src/pybind/mgr/dashboard/frontend/src/app/shared/models/cd-table-action.ts

```diff
@@ -9,7 +9,7 @@ export class CdTableAction {
   // This is the function that will be triggered on a click event if defined
   click?: Function;
 
-  permission: 'create' | 'update' | 'delete' | 'read';
+  permission: 'create' | 'update' | 'delete' | 'read' | boolean;
```

Comment:
This feels really odd to me. By mixing these type (which I think I've refactored in the past to not be mixed up), introduces a few unattractivenesses in the code above. It requires us to check for the type every time we want to access the `permission` attribute, like these:

```
      ['update', 'delete'].includes(permission as string) && (!selected || selected.cdExecuting)
```

```
    if (_.isBoolean(permission)) {
```

```
      return _.isBoolean(perm) ? perm : permissions.includes(perm as string);
```

It also introduces permissions like `trueAction` and `falseAction`, which I personally don't find easy to grasp, especially when considering the other actions in the same array: `addAction`, `editAction`, `unprotectAction`.

Is there really no other way to implement this?

I'd be in favor of finding a cleaner solution here.